### DISCUSSION TITLE
My website ran into an error. Please Help!

### DISCUSSION THREAD

**Discussion by ghost** [community/community#58395] [Programming Help] [answered]
### Body

if(!localStorage.getItem("username")){
  alert("Redirecting you to the signup page");
  window.location.assign("signup");
}
function change_username(){
  var username = prompt("Enter your new username > ");
  localStorage.removeItem("username");
  localStorage.setItem("username", username);
  var p = document.createElement("p"):
  p.innerText = "Your username has been saved.";
  p.style.color = "green";
  p.style.align = "center";
  document.body.appendChild(p);
  var thanksNote = document.createElement("footer");
  thanksNote.innerHTML = "<footer>Thanks for using <em>PyTutorials</em></footer>";
  thanksNote.style.position = "absolute";
  thanksNote.style.align = "center";
  thanksNote.top = "100%";
  
}
function remove_username(){
  localStorage.removeItem("username");
  alert("You have been signed out! Refresh the page to see the changes");
}
function dont_username_popup(){
  if(!localStorage.getItem("popup_username=false")){
    localStorage.setItem("popup_username=false", false);
    var p = document.createElement("p");
    p.innerText = "Changes Saved!";
    p.style.align = "center";
    p.style.color = "green";
    document.body.appendChild(p);
    var thanksNote = document.createElement("footer");
    thanksNote.innerHTML = "<footer>Thanks for using <em>PyTutorials</em></footer>";
    thanksNote.style.position = "absolute";
    thanksNote.style.top = "100%":
    thanksNote.style.align = "center":
    document.body.appendChild(thanksNote);
  }else{
    var p = document.createElement("p");
    p.innerText = "This is already set!";
    p.style.align = "center";
    p.style.color = "red";
    var thanksNote = document.createElement("footer");
    thanksNote.innerHTML = "<footer>Thank you for using <em>PyTutorials</em></footer>";
    thanksNote.style.align = "center";
    thanksNote.style.positon = "absolute";
    thanksNote.style.top = "100%";
    document.body.appendChild(thanksNote);
    document.body.appendChild(p);
 }

 https://GitHub.com/pytutorials/pytutorials.github.io/username.js

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ghost** [comment]
Sorry, the link is at https://github.com/pytutorials/pytutorials.github.io/blob/main/username.js

**Reply by palonza** [reply]
hi @pytutorials , what mistake do you have? 1) copy and paste a screenshot 2) show browser console output 3) tell us what you are trying to do

**Comment by ghost** [comment]
I’m using IOS so I can’t view the console. But if you go to the page https://pytutorials.github.io/username it doesn’t work. My HTML code is https://github.com/pytutorials/pytutorials.github.io/blob/main/username.html

**Reply by ghost** [reply]
The code is also at https://codepen.io/Bob57/pen/wvQGOVB?editors=1111

**Reply by palonza** [reply]
hi @pytutorials ,

I'm sorry to tell you that it will never work, the reason is the code that you publish on github need to be static, for dynamic programmatic code will never work for you, it must be hosted on some web server: 1) apache, 2) nodejs, etc., in another way too, you could integrate it with external services: 1) firebase, 2) heroku.

A word of advice, you could first start by publishing your codes locally on your machine or using codepen.io, but never github.

for more information about how publish site, you can follow this from official github:
[https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages).

I have also noticed that the javascript code in username.js contains several syntax errors. But first you must know how to publish code. I suggest you read tutorials on the internet or videos on youtube.

I hope I could help you.

Please, mark your question as answered. Thanks in advance.

**Reply by ghost** [reply]
It is publish on GitHub I was just using a website that allowed you to view the console.

**Reply by palonza** [reply]
Ok... I will review...

**Reply by palonza** [reply]
give me a few minutes

**Comment by palonza** [comment, marked as answer]
Hi @pytutorials

Please check: [https://jsfiddle.net/polook/1m0rz2bf/51/](https://jsfiddle.net/polook/1m0rz2bf/51/)

- I have simplified your code and made it more readable.
- I have removed the redirection since it does not work in pages like codepen and fiddler.
- I have changed the anchor elements to button elements.
- I've removed the createElement and replaced it with getElementById to make it prettier.

I hope this code helps you in understanding javascript. Remember to indent and format your code to improve your understanding. I suppose that it is a learning exercise, since a cookie should **never be used for secure login**, for a productive environment, they must invest several days and above all add more technologies mentioned above.

Please mark your question as answered to help me.

Thanks since now.

**Reply by queenofcorgis** [reply]
Nice work @palonza 🤩 What a great reply/answer and thank you @pytutorials for marking the answer ❤️

**Comment by martinsolod** [comment]
Hi @palonza, the code bellow is more readable. One question: Tell me, how can I make this source code work on a real web page? That I need?

**Reply by palonza** [reply]
Hi @martinsolod, I mentioned above: do you need reactjs or angular in a frontend side and nodejs in backend side, futhermore, a saas or paas technology someone like a azure, aws or gcp.
